backend/src/main/java/be/cytomine/appstore/services/TaskService.java

```diff
@@ -0,0 +1,490 @@
+package be.cytomine.appstore.services;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+
+import be.cytomine.appstore.dto.handlers.filestorage.Storage;
+import be.cytomine.appstore.dto.handlers.registry.DockerImage;
+import be.cytomine.appstore.dto.inputs.task.TaskAuthor;
+import be.cytomine.appstore.dto.inputs.task.TaskDescription;
+import be.cytomine.appstore.dto.inputs.task.UploadTaskArchive;
+import be.cytomine.appstore.dto.misc.TaskIdentifiers;
+import be.cytomine.appstore.dto.responses.errors.AppStoreError;
+import be.cytomine.appstore.dto.responses.errors.ErrorBuilder;
+import be.cytomine.appstore.dto.responses.errors.ErrorCode;
+import be.cytomine.appstore.exceptions.BundleArchiveException;
+import be.cytomine.appstore.exceptions.FileStorageException;
+import be.cytomine.appstore.exceptions.RegistryException;
+import be.cytomine.appstore.exceptions.TaskNotFoundException;
+import be.cytomine.appstore.exceptions.TaskServiceException;
+import be.cytomine.appstore.exceptions.ValidationException;
+import be.cytomine.appstore.handlers.RegistryHandler;
+import be.cytomine.appstore.handlers.StorageData;
+import be.cytomine.appstore.handlers.StorageHandler;
+import be.cytomine.appstore.models.CheckTime;
+import be.cytomine.appstore.models.Match;
+import be.cytomine.appstore.models.task.Author;
+import be.cytomine.appstore.models.task.Parameter;
+import be.cytomine.appstore.models.task.ParameterType;
+import be.cytomine.appstore.models.task.Task;
+import be.cytomine.appstore.models.task.TypeFactory;
+import be.cytomine.appstore.repositories.TaskRepository;
+import be.cytomine.appstore.utils.ArchiveUtils;
+import com.fasterxml.jackson.databind.JsonNode;
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+@Slf4j
+@RequiredArgsConstructor
+@Service
+public class TaskService
+{
+
+    private final TaskRepository taskRepository;
+
+    private final StorageHandler fileStorageHandler;
+
+    private final RegistryHandler registryHandler;
+
+    private final TaskValidationService taskValidationService;
+
+    private final ArchiveUtils archiveUtils;
+
+    @Value("${storage.input.charset}")
+    private String charset;
+
+    @Value("${scheduler.task-resources.ram}")
+    private String defaultRam;
+
+    @Value("${scheduler.task-resources.cpus}")
+    private int defaultCpus;
```

Comment:
Since this is the app store and not for execution, remove this since there is no need for default values